lib/gateway/server.js

```diff
@@ -20,9 +20,16 @@ function createTlsServer (httpsConfig, app) {
   let defaultCert = null;
   let sniCerts = [];
 
-  for (let el in httpsConfig.tls) {
-    let domain = el;
-    let certPaths = httpsConfig.tls[el];
+  let domainCount = httpsConfig.tls.length;
+  let domainNames = [];
+
+  httpsConfig.tls.forEach(domainObj => {
```

Comment:
@kevinswiber great code condensing suggestion. 

CircleCI is showing quite a few errors and warnings with my pull request. Is this all caused by a disruption from this change?

Also, if no `tls` option is set in `gateway.config.yml` for https i.e.
```yml
https:
  port: 9443
```
then `npm start` silently fails to begin listening on the specified port allowing the process to continue with no error output. 

Should an error be thrown in this file to make a developer aware of the potential bug?